test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala

```diff
@@ -0,0 +1,185 @@
+package scala
+package tools.nsc
+package reporters
+
+import java.io.{ByteArrayOutputStream, StringReader, BufferedReader, PrintStream, PrintWriter}
+import org.junit.Assert._
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+import scala.tools.nsc.Settings
+import scala.tools.nsc.reporters.ConsoleReporter
+import scala.reflect.internal.util._
+import scala.reflect.io.AbstractFile
+
+
+@RunWith(classOf[JUnit4])
+class ConsoleReporterTest {
+  @Test
+  def unitTest: Unit = {
+    val noPos = NoPosition
+    val path = "test/junit/scala/tools/nsc/reporters/file_ConsoleReporterTest.txt"
+    val file = AbstractFile getFile(path)
+    val batch_file = new BatchSourceFile(file)
+    val posWithFile = new OffsetPosition(batch_file, 4)
+    val content = posWithFile.lineContent
+
+    val err_out = new ByteArrayOutputStream  //for storing the output printed in the ConsoleReporter
+    val output_writer = new PrintWriter(err_out)
+    val input = new StringReader("s\n")
+    val input_reader = new BufferedReader(input)
+    val setting = new Settings()
+    //Changing maxerrs and maxwanrs
+    setting.maxerrs.tryToSet(List("50"))
+    setting.maxwarns.tryToSet(List("25"))
+    val reporter = new ConsoleReporter(setting, input_reader, output_writer)
+
+
+    // printMessage test
+    reporter.printMessage("Hello World!")
+    assertEquals("Hello World!\n", err_out.toString)
+    err_out.reset
```

Comment:
I'm not sure that I get you correctly. Do you want me to group the related tests inside a method, for example the tests for printMessage inside a single method?